pkg/controller/v1alpha2/core/scopes/healthscope/healthscope.go

```diff
@@ -205,12 +232,55 @@ func CheckDaemonsetHealth(ctx context.Context, client client.Client, ref runtime
 		r.Diagnosis = errors.Wrap(err, errHealthCheck).Error()
 		return r
 	}
-	r.Target.UID = daemonset.GetUID()
+	r.ComponentName = getComponentNameFromLabel(&daemonset)
+	r.TargetWorkload.UID = daemonset.GetUID()
+	r.Diagnosis = fmt.Sprintf(infoFmtReady, daemonset.Status.NumberReady, daemonset.Status.DesiredNumberScheduled)
 
+	// Health criteria
 	if daemonset.Status.NumberUnavailable != 0 {
-		r.Diagnosis = fmt.Sprintf(errFmtResourceNotReady, daemonset.Status)
 		return r
 	}
-	r.IsHealthy = true
+	r.HealthStatus = StatusHealthy
 	return r
 }
+
+// CheckByHealthCheckTrait checks health condition through HealthCheckTrait.
+func CheckByHealthCheckTrait(ctx context.Context, c client.Client, wlRef runtimev1alpha1.TypedReference, ns string) *WorkloadHealthCondition {
+	//TODO(roywang) implement HealthCheckTrait feature
+	return nil
+}
+
+// CheckUnknownWorkload handles unknown type workloads.
+func CheckUnknownWorkload(ctx context.Context, c client.Client, wlRef runtimev1alpha1.TypedReference, ns string) *WorkloadHealthCondition {
```

Comment:
I didn't see any test case for this function?

Reply:
It's tested [here](https://github.com/captainroy-hy/oam-kubernetes-runtime/blob/8b62ad51025053c63c2e682d71cc8d43487113be/pkg/controller/v1alpha2/core/scopes/healthscope/healthscope_controller_test.go#L269). Let me separate it into healthscope_test.go